src/ViewLayer/Faults/BatteryFaults/BatteryFaultList.cpp

```diff
@@ -32,13 +32,20 @@ FaultDisplayData BatteryFaultList::nextActiveFault()
     return FaultDisplayData();
 }
 
+QVector<QString>& BatteryFaultList::activeFaultLabels()
+{
+    numberOfActiveFaults();
+    return activeLabels_;
+}
+
 QMap<BatteryFaults, FaultDisplayData>& BatteryFaultList::faults()
 {
     return faultList_;
 }
 
-int BatteryFaultList::numberOfActiveFaults() const
+int BatteryFaultList::numberOfActiveFaults()
 {
+    activeLabels_.clear();
     int numberOfActiveFaults = 0;
 
     QMap<BatteryFaults, FaultDisplayData>::const_iterator i ;
```

Comment:
Can we change the for loop into something more consise? 
Ex. std::for_each(faultList_.constBegin(), faultList_.constEnd(), function here), or std::transform (std::transform probs doesn't work here, but just wanted to push the idea of using std)

Reply:
I'm not sure if there is something like that that exists.